service/module.go

```diff
@@ -108,47 +106,53 @@ type moduleWrapper struct {
 func newModuleWrapper(
 	host Host,
 	moduleProvider ModuleProvider,
-	options ...ModuleOption,
+	options ...ModuleOptionFn,
 ) (*moduleWrapper, error) {
 	if moduleProvider == nil {
 		return nil, nil
 	}
-	moduleOptions := &moduleOptions{}
+	moduleOption := &moduleOption{}
 	for _, option := range options {
-		if err := option(moduleOptions); err != nil {
+		if err := option(moduleOption); err != nil {
 			return nil, err
 		}
 	}
-	if moduleOptions.name == "" {
-		moduleOptions.name = moduleProvider.DefaultName()
+	var name string
+	var roles []string
+	if moduleOption.ServiceName != "" {
+		name = moduleOption.ServiceName
+	} else {
+		name = host.Name()
 	}
-	scopedHost, err := newScopedHost(
-		host,
-		moduleOptions.name,
-		moduleOptions.roles...,
-	)
-	if err != nil {
-		return nil, err
+	if len(moduleOption.Roles) > 0 {
+		roles = moduleOption.Roles
+	} else {
+		roles = host.Roles()
 	}
+	scopedHost := &scopedHost{Host: host, name: name, roles: roles}
 	module, err := moduleProvider.Create(scopedHost)
 	if err != nil {
 		return nil, err
 	}
 	if module == nil {
 		return nil, nil
 	}
-	return &moduleWrapper{module: module, scopedHost: scopedHost}, nil
+	return &moduleWrapper{
+		name:       reflect.TypeOf(module).PkgPath(),
```

Comment:
Another way to solve this is bring back the Name() function on the Module interface. That's more explicit and clear rather than picking up the package name via reflect. Thoughts?